CMIP7/esm1p6/atmosphere/nitrogen/cmip7_PI_nitrogen_generate.py

```diff
@@ -0,0 +1,109 @@
+from argparse import ArgumentParser
+from pathlib import Path
+
+import iris
+from cmip7_ancil_argparse import common_parser
+from cmip7_ancil_common import (
+    INTERPOLATION_SCHEME,
+    esm_grid_mask_cube,
+    fix_coords,
+    save_ancil,
+)
+from cmip7_ancil_constants import ANCIL_TODAY
+from iris.util import equalise_attributes
+from nitrogen.cmip7_nitrogen import (
+    NITROGEN_SPECIES,
+    NITROGEN_STASH_ITEM,
+    cmip7_nitrogen_dirpath,
+)
+
+
+def parse_args():
+    parser = ArgumentParser(
+        parents=[common_parser()],
+        prog="cmip7_PI_nitrogen_generate",
+        description=(
+            "Generate input files from CMIP7 pre-industrial nitrogen forcings"
+        ),
+    )
+    parser.add_argument("--dataset-date-range")
+    parser.add_argument("--save-filename")
+    return parser.parse_args()
+
+
+def cmip7_pi_nitrogen_filepath(args, species):
+    dirpath = cmip7_nitrogen_dirpath(args, "monC", species)
+    filename = (
+        f"{species}_input4MIPs_surfaceFluxes_CMIP_"
+        f"{args.dataset_version}_gn_"
+        f"{args.dataset_date_range}-clim.nc"
+    )
+    return dirpath / filename
+
+
+def load_cmip7_pi_nitrogen_species(args, species):
+    species_filepath = cmip7_pi_nitrogen_filepath(args, species)
+    cube = iris.load_cube(species_filepath)
+    return cube
+
+
+def load_cmip7_pi_nitrogen(args):
+    # Load all of the PI nitrogen datasets into a CubeList
+    nitrogen_cubes = iris.cube.CubeList()
+    for species in NITROGEN_SPECIES:
+        species_cube = load_cmip7_pi_nitrogen_species(args, species)
+        nitrogen_cubes.append(species_cube)
+    # Remove all attributes that differ between cubes
+    equalise_attributes(nitrogen_cubes)
+    # Add the cubes together
+    cube_tot = nitrogen_cubes[0].copy()
+    for cube_nbr in range(1, 4):
+        iris.analysis.maths.add(
+            cube_tot, nitrogen_cubes[cube_nbr], in_place=True
+        )
```

Comment:
Here you are summing all values of each cube in the cubelist together, and you are only interested in the total sum. For this reason, there's no need to copy the first cube in the cubelist, you can use it directly.
I would simplify with:
```suggestion
    # Add the cubes together
    cube_tot = nitrogen_cubes[0]
    for cube in nitrogen_cubes[1:]:
        cube_tot = cube_tot + cube
```

Reply:
Will do.